com.unity.multiplayer.mlapi/Runtime/Messaging/MessageBatcher.cs

```diff
@@ -126,40 +137,55 @@ public void QueueItem(in RpcFrameQueueItem queueItem)
                 if (sendStream.IsEmpty)
                 {
                     sendStream.IsEmpty = false;
-                    sendStream.NetworkChannel = queueItem.NetworkChannel;
-
-                    switch (queueItem.QueueItemType)
-                    {
-                        // 8 bits are used for the message type, which is an NetworkConstants
-                        case RpcQueueContainer.QueueItemType.ServerRpc:
-                            sendStream.Writer.WriteByte(NetworkConstants.SERVER_RPC); // MessageType
-                            break;
-                        case RpcQueueContainer.QueueItemType.ClientRpc:
-                            sendStream.Writer.WriteByte(NetworkConstants.CLIENT_RPC); // MessageType
-                            break;
-                    }
+                    sendStream.NetworkChannel = item.NetworkChannel;
+                }
+                // If the item is a different channel we have to flush and change channels.
+                // This isn't great if channels are interleaved, but having a different stream
+                // per channel would create ordering issues.
+                else if (sendStream.NetworkChannel != item.NetworkChannel)
+                {
+                    sendCallback(clientId, sendStream);
+                    // clear the batch that was sent from the SendDict
+                    sendStream.Buffer.SetLength(0);
+                    sendStream.Buffer.Position = 0;
+                    ProfilerStatManager.MessageBatchesSent.Record();
```

Comment:
@becksebenius-unity dunno if this 'channel context thing' is a big enough deal but would you want a metric for "interrupted batching from channel switches"?